src/MongoDB.Driver.Core/Core/Operations/RetryabilityHelper.cs

```diff
@@ -83,16 +83,19 @@ public static bool IsResumableChangeStreamException(Exception exception)
 
         public static bool IsRetryableWriteException(Exception exception)
         {
+            if (__retryableWriteExceptions.Contains(exception.GetType()))
```

Comment:
Do this check first now, and only fall through if not returning true from just this check.